core/src/main/java/edu/wpi/grip/core/sockets/InputSocket.java

```diff
@@ -114,7 +114,14 @@ public void onValueChanged() {
 
     @Override
     public boolean equals(Object o) {
-      return o instanceof InputSocket && decorated.equals(o);
+      if (o instanceof InputSocket) {
+        if (decorated.equals(o)) {
+          return true;
+        } else if (o instanceof Decorator) {
+          return decorated.equals(((Decorator) o).decorated);
+        }
```

Comment:
This could be written as a single line return: 
```
return decorated.equals(o) || (o instanceof Decorator && decorated.equals(((Decorator) o).decorated));
```